build.zig

```diff
@@ -65,12 +65,15 @@ pub fn build(b: *std.Build) !void {
         []const u8,
         "git-commit",
         "The git commit revision of the source code.",
-    ) orelse try shell.git_commit();
-    options.addOption([]const u8, "git_commit", git_commit);
+    ) orelse std.mem.trimRight(u8, b.exec(&.{ "git", "rev-parse", "--verify", "HEAD" }), "\n");
+    assert(git_commit.len == 40);
+    options.addOption(?[40]u8, "git_commit", git_commit[0..40].*);
 
-    const release_version_string = b.env_map.get("TIGERBEETLE_RELEASE");
-    options.addOption([]const u8, "version", release_version_string orelse try shell.git_tag());
-    options.addOption([]const u8, "release", release_version_string orelse "0.0.1");
+    options.addOption(
+        []const u8,
+        "release",
+        b.option([]const u8, "tigerbeetle-release", "Release triple.") orelse "0.0.1",
+    );
```

Comment:
Aren't all of these options "tigerbeetle" options? Maybe we should just call this `"version"`, to avoid confusion with `-Drelease`/release-builds.

Reply:
Yeah, I am undecided here. We use `release` internally, but `-Drelease` is already occupied by, well, `-Drelease`.

I settled on `tigerbeetle-release` because its a long ugly hyphen-word --- this should only be set by the release infra, so verbose is good I guess. 

No strong opinion either way.

Reply:
Just thought of another option: `-Dconfig-release` (to match `-Dconfig-log-level`, `-Dconfig-aof-record` etc).

Reply:
Excellent! https://github.com/tigerbeetle/tigerbeetle/pull/1811